sw/device/tock/chips/opentitan_common/src/uart.rs

```diff
@@ -127,90 +40,89 @@ impl Uart<'a> {
         }
     }
 
-    fn set_baud_rate(&self, baud_rate: u32) {
-        let regs = self.registers;
-        let uart_ctrl_nco = ((baud_rate as u64) << 20) / self.clock_frequency as u64;
-
-        regs.ctrl
-            .write(ctrl::nco.val((uart_ctrl_nco & 0xffff) as u32));
-        regs.ctrl.modify(ctrl::tx::SET + ctrl::rx::SET);
-
-        regs.fifo_ctrl
-            .write(fifo_ctrl::rxrst::SET + fifo_ctrl::txrst::SET);
+    fn interrupt_enable(
+        &self,
+        interrupt: dif_uart_interrupt_t,
+        enable: dif_uart_enable_t,
+    ) -> ReturnCode {
+        self.dif_uart.map_or(ReturnCode::FAIL, |uart| unsafe {
+            match dif_uart_irq_enable(uart, interrupt, enable) {
+                kDifUartOk => ReturnCode::SUCCESS,
+                _ => ReturnCode::FAIL,
+            }
+        })
     }
 
-    fn enable_tx_interrupt(&self) {
-        let regs = self.registers;
-
-        regs.intr_enable.modify(intr::tx_empty::SET);
+    fn bytes_send(&self, data: *const u8, num_to_send: usize) -> Result<usize, ReturnCode> {
+        self.dif_uart.map_or(Err(ReturnCode::FAIL), |uart| unsafe {
+            let mut bytes_sent: usize = 0;
+            let bytes_sent_p: *mut usize = &mut bytes_sent;
```

Comment:
It didn't work before when I tried to inline it, I don't remember what it was complaining about. I will have a go, and let you know.

Reply:
Unfortunately this `dif_uart_bytes_send(uart, data, num_to_send, **bytes_sent as *mut usize**)` doesn't work. When converting it in this way, it casts the value `bytes_sent` holds, rather than the address of the variable.

I think the cast I have is pretty standard way of casting in Rust.